src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -285,7 +277,9 @@ def _callback(self, header, err):
 
     def _parse_file_name(self, path):
         matched = parse_file_path(self.file_regex, path)
-        data_source = get_entry(matched, "data_source", self.data_source)
+        data_source = self.data_source
```

Comment:
This has now a different meaning. If both, `self.data_source` and `matched["data_source"]` are present, which one should win?

Reply:
What do we want? In the original version one given in config will overwrite one exctacted from file path.

Reply:
If one puts in the effort to add a `data_source` group to the regex and it is ignored, I think it would be surprising. Maybe rename the config option to `default_data_source` to make it more clear.

Reply:
Currently this works opposite way. If one have data_source in the config file it will overwrite one found in the regexp.

Reply:
Alternatively we can use `default_data_source` as you suggest. This will be used only if data_source is not found in the regexp.